modules/prebidServerBidAdapter/index.js

```diff
@@ -105,45 +105,45 @@ config.setDefaults({
 });
 
 /**
- * @param {S2SConfig} option
+ * @param {S2SConfig} s2sConfig
  * @return {boolean}
  */
-function updateConfigDefaultVendor(option) {
-  if (option.defaultVendor) {
-    let vendor = option.defaultVendor;
-    let optionKeys = Object.keys(option);
+function updateConfigDefaults(s2sConfig) {
+  if (s2sConfig.defaultVendor) {
+    let vendor = s2sConfig.defaultVendor;
+    let optionKeys = Object.keys(s2sConfig);
     if (S2S_VENDORS[vendor]) {
       // vendor keys will be set if either: the key was not specified by user
       // or if the user did not set their own distinct value (ie using the system default) to override the vendor
       Object.keys(S2S_VENDORS[vendor]).forEach((vendorKey) => {
-        if (s2sDefaultConfig[vendorKey] === option[vendorKey] || !includes(optionKeys, vendorKey)) {
-          option[vendorKey] = S2S_VENDORS[vendor][vendorKey];
+        if (s2sDefaultConfig[vendorKey] === s2sConfig[vendorKey] || !includes(optionKeys, vendorKey)) {
+          s2sConfig[vendorKey] = S2S_VENDORS[vendor][vendorKey];
         }
       });
     } else {
       logError('Incorrect or unavailable prebid server default vendor option: ' + vendor);
       return false;
     }
+  } else {
+    if (s2sConfig.adapter == null) {
```

Comment:
Curious why this is `null` and not `undefined` . Would a truthy check maybe better, e.g `if(s2sConfig)` .

Reply:
`== null` only double equals, "[loose equality](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness)" means "is null or  undefined", so switching this to `undefined` would have the same effect but for more keystrokes. 

A truth check (I assume you meant `if (!s2sConfig.adapter)`) would also pick up a laundry list of stuff that javascript considers "[falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy)". None of them would make sense here so it would be fine, but I'd argue not really better, since an error would make more sense than an override for most of them IMO.

Reply:
Make sense! I haven't sense `==` in a while and must have misread as `===` . This is quite neat!